altgraph/__init__.py

```diff
@@ -139,9 +139,15 @@
   @contributor: U{Reka Albert <http://www.phys.psu.edu/~ralbert/>}
 
 """
-import pkg_resources
+try:
+    import pkg_resources
 
-__version__ = pkg_resources.require("altgraph")[0].version
+    __version__ = pkg_resources.require("altgraph")[0].version
+except ImportError:
+    # pkg_resources was killed in Python3.12
+    import importlib_metadata
+
```

Comment:
These should be reversed so we try the new way and then fall back to the old way.  Also, there is a difference between `importlib_metadata` and `importlib.metadata`.  I believe that the better approach is
* #25